sway/desktop/output.c

```diff
@@ -922,9 +912,26 @@ void handle_new_output(struct wl_listener *listener, void *data) {
 		output_repaint_timer_handler, output);
 
 	struct output_config *oc = find_output_config(output);
+	
 	apply_output_config(oc, output);
 	free_output_config(oc);
 
+	/* TODO read from config for leasing new outputs */
+	if (wlr_output->non_desktop) {
+		if (output->enabled) {
+			sway_log(SWAY_DEBUG, "Disabling non-desktop output");
+			output_disable(output);
+			wlr_output_layout_remove(root->output_layout, wlr_output);
+		}
```

Comment:
Hm it's a bit unfortunate to do this back-and-forth. I think with my behavior suggestion this will go away anyways.